cli/dcoscli/auth/main.py

```diff
@@ -0,0 +1,109 @@
+import dcoscli
+import docopt
+from dcos import cmds, config, emitting, http, util
+from dcos.errors import DCOSAuthorizationException, DCOSException
+from dcoscli.subcommand import default_command_info, default_doc
+from dcoscli.util import decorate_docopt_usage
+
+from six.moves import urllib
+
+emitter = emitting.FlatEmitter()
+logger = util.get_logger(__name__)
+
+
+def main(argv):
+    try:
+        return _main(argv)
+    except DCOSException as e:
+        emitter.publish(e)
+        return 1
+
+
+@decorate_docopt_usage
+def _main(argv):
+    args = docopt.docopt(
+        default_doc("auth"),
+        argv=argv,
+        version='dcos-auth version {}'.format(dcoscli.version))
+
+    http.silence_requests_warnings()
+
+    return cmds.execute(_cmds(), args)
+
+
+def _cmds():
+    """
+    :returns: all the supported commands
+    :rtype: list of dcos.cmds.Command
+    """
+
+    return [
+        cmds.Command(
+            hierarchy=['auth', 'login'],
+            arg_keys=[],
+            function=_login),
+
+        cmds.Command(
+            hierarchy=['auth', 'logout'],
+            arg_keys=[],
+            function=_logout),
+
+        cmds.Command(
+            hierarchy=['auth'],
+            arg_keys=['--info'],
+            function=_info),
+    ]
+
+
+def _info(info):
+    """
+    :param info: Whether to output a description of this subcommand
+    :type info: boolean
+    :returns: process status
+    :rtype: int
+    """
+
+    emitter.publish(default_command_info("auth"))
+    return 0
+
+
+def _login():
+    """
+    :returns: process status
+    :rtype: int
+    """
+
+    conf = util.get_config()
+    token = conf.get("core.dcos_acs_token")
+    if token:
+        raise DCOSException("You are already logged in")
```

Comment:
We should only do this if we know that the token doesn't return 401, no? I think what we want to do is remember that the token was already set and change line 95 to check this value and print the right message. What do you think?